bucket/kafka.json

```diff
@@ -0,0 +1,49 @@
+{
+    "version": "2.2.0",
+    "description": "Apache Kafka. A distributed streaming platform.",
+    "homepage": "https://kafka.apache.org/",
+    "license": "Apache-2.0",
+    "url": "https://www.apache.org/dist/kafka/2.2.0/kafka_2.12-2.2.0.tgz",
+    "hash": "a009624fae678fa35968f945e18e45fbea9a30fa8080d5dcce7fdea726120027",
+    "extract_dir": "kafka_2.12-2.2.0",
+    "bin": [
+        "bin\\windows\\connect-distributed.bat",
+        "bin\\windows\\connect-standalone.bat",
+        "bin\\windows\\kafka-acls.bat",
+        "bin\\windows\\kafka-broker-api-versions.bat",
+        "bin\\windows\\kafka-configs.bat",
+        "bin\\windows\\kafka-console-consumer.bat",
+        "bin\\windows\\kafka-console-producer.bat",
+        "bin\\windows\\kafka-consumer-groups.bat",
+        "bin\\windows\\kafka-consumer-perf-test.bat",
+        "bin\\windows\\kafka-delegation-tokens.bat",
+        "bin\\windows\\kafka-dump-log.bat",
+        "bin\\windows\\kafka-mirror-maker.bat",
+        "bin\\windows\\kafka-preferred-replica-election.bat",
+        "bin\\windows\\kafka-producer-perf-test.bat",
+        "bin\\windows\\kafka-reassign-partitions.bat",
+        "bin\\windows\\kafka-replica-verification.bat",
+        "bin\\windows\\kafka-run-class.bat",
+        "bin\\windows\\kafka-server-start.bat",
+        "bin\\windows\\kafka-server-stop.bat",
+        "bin\\windows\\kafka-topics.bat",
+        "bin\\windows\\zookeeper-server-start.bat",
+        "bin\\windows\\zookeeper-server-stop.bat",
+        "bin\\windows\\zookeeper-shell.bat"
+    ],
+    "persist": "config",
+    "suggest": {
+        "JDK": [
+            "java/oraclejdk",
+            "java/openjdk"
+        ]
+    },
+    "checkver": {
+        "url": "https://kafka.apache.org/downloads",
+        "regex": "current stable version is ([\\d.]+)\\."
+    },
+    "autoupdate": {
+        "url": "https://www.apache.org/dist/kafka/$version/kafka_2.12-$version.tgz",
+        "extract_dir": "kafka_2.12-$version"
```

Comment:
Add hash extraction

```suggestion
        "hash": {
            "url": "$url.sha512"
        },,
        "extract_dir": "kafka_2.12-$version"
```

![A](https://i.imgur.com/lk5KIfD.png)

Reply:
I tried with the `.sha1` file but that didn't work.
My guess is that the format used in that file is not supported in scoop:
```
kafka_2.12-2.2.0.tgz: 360B 39BD 5B7C D259 A47D  D277 966C 3314 0EFF FB8C
```
I'll try with the `.sha512` version as well, but that one looks similar:
```
kafka_2.12-2.2.0.tgz: 63FDCD05 B857A096 D2E72804 253590BD 086E4F30 46FBFA54
                      6F186602 E48F5471 2ED6DE2A 346A5B05 D5335870 BF5268FC
                      B2BFD3EF 9C4F4BBE A379091A 79604556
```

Reply:
Nope, same problem:
```
.\bin\checkver.ps1 kafka -u
kafka: 2.2.0 (scoop version is 2.1.0) autoupdate available
Autoupdating kafka
Searching hash for kafka_2.12-2.2.0.tgz in https://www.apache.org/dist/kafka/2.2.0/kafka_2.12-2.2.0.tgz.sha512
Could not find hash in https://www.apache.org/dist/kafka/2.2.0/kafka_2.12-2.2.0.tgz.sha512
Downloading kafka_2.12-2.2.0.tgz to compute hashes!
Loading kafka_2.12-2.2.0.tgz from cache
Computed hash: a009624fae678fa35968f945e18e45fbea9a30fa8080d5dcce7fdea726120027
```

The constructed URL is correct, but scoop doesn't understand the (rather strange) notation in the file.

Reply:
Will work on a fix for removing the linebreaks from the hash.

The `.sha1` file keeps everything in a single line and it can be matched with this: 

```json
        "hash": {
            "url": "$url.sha1",
            "regex": "$basename:[\\x20\\t]+([a-fA-F0-9\\s]+)\\n+"
        }
```

Reply:
@r15ch13 that worked, thanks for the tip!
Amended to the original commit.
@Ash258 good to merge?